costmodel/promparsers.go

```diff
@@ -148,6 +148,11 @@ func parseDataPoint(dataPoint interface{}) (*Vector, error) {
 		return nil, err
 	}
 
+	if math.IsNaN(v) {
+		klog.V(1).Infof("Warning: Found NaN value parsing vector data point.")
+		v = 0.0
```

Comment:
thought you wanted to not return 0? change your mind?

Reply:
Well, I started looking later in the path and noticed we do a lot of checking for 0 when it can be dangerous. Since, this happens early on, I think it's safe to rely on our 0 checking later.